tests/test_integration/test_common_layout.py

```diff
@@ -0,0 +1,143 @@
+    """Check that the components common to all pages are created.
+
+    The components common to all pages are the page content container and
+    the sidebar.
+
+    Parameters:
+        dash_duo : DashComposite
+            Default fixture for Dash Python integration tests.
+        timeout : float
+            maximum time to wait in seconds for a component
+    """
+
+    # start server
+    dash_duo.start_server(app)
+
+    # wait for main content to be rendered
+    try:
+        dash_duo.wait_for_element("#page-content", timeout=timeout)
+    except selenium.common.exceptions.TimeoutException:
+        pytest.fail("Main content component not generated")
+
+    # wait for sidebar to be rendered
+    try:
+        dash_duo.wait_for_text_to_equal(
+            "#sidebar h2", "WAZP 🐝", timeout=timeout
+        )
+    except selenium.common.exceptions.TimeoutException:
+        pytest.fail("Sidebar component not generated")
+
+    # check there are no errors in browser console
+    assert (
+        dash_duo.get_logs() == []
+    ), "There are errors in the browser console!"
+
+
+@pytest.mark.xfail(
+    raises=AssertionError,
+    reason=(
+        "Feature not yet implemented:"
+        "When config has not been loaded, "
+        "warnings should show in pages that are not Home"
+    ),
+    strict=True,
+    # with strict=True
+    # if the test passes unexpectedly,
+    # it will fail the test suite
+)
+def test_sidebar_links(
+    dash_duo: DashComposite,
+    map_page_name_to_title: dict,
+    timeout: float,
+) -> None:
+    """Check the sidebar links take to pages with the expected title
+    and that no errors occur in the browser console
+
+    Parameters:
+        dash_duo : DashComposite
+            Default fixture for Dash Python integration tests.
+        map_page_name_to_title: dictionary : dict
+            dictionary with page names as keys, and page titles as values
+        timeout : float
+            maximum time to wait in seconds for a component
+    """
+
+    # start server
+    dash_duo.start_server(app)
+
+    # click through links in sidebar
+    for page in dash.page_registry.values():
+
+        # click thru each page
+        dash_duo.find_element(
+            "#sidebar #link-" + page["name"].replace(" ", "-"),
+        ).click()
+
+        # check page title
+        try:
+            dash_duo.wait_for_text_to_equal(
+                "h1", map_page_name_to_title[page["name"]], timeout=timeout
+            )
+
+        except selenium.common.exceptions.TimeoutException:
+            pytest.fail(
+                f'Timeout waiting for page {page["name"]} '
+                "to show a title with the text: "
+                f'{map_page_name_to_title[page["name"]]}'
+            )
+
+        # click back to home
+        # Q for review: I do this to make the starting point consistent...
+        # but is it required? should I use fixtures instead?
+        dash_duo.find_element("#sidebar #link-Home").click()
```

Comment:
```suggestion
```

Ah. So what I did was [parametrise the test and go to each page](https://github.com/SainsburyWellcomeCentre/WAZP/pull/61/files#diff-0c753256829d1762d9470d207b219335bcc1f4ceb060b490c44b3f5f50b14fc5R10-R19). That way we always start from the homepage with a fresh server.

I don't have a strong feeling if it makes sense to go back each time. Maybe... not?... because that's not what a user would do. So, _very_ soft suggestion: delete this.